### ISSUE TITLE
ImageNet pretrained Aligned Xception model

### ISSUE THREAD

**Issue by PkuRainBow** [Gaoyiminggithub/Graphonomy#8]
Thanks for sharing the repo.

I notice that another [repo](https://github.com/jfzhang95/pytorch-deeplab-xception) mentions that there exist no ImageNet pre-trained Aligned Xception model, and the performance is worse than the ResNet-101.

In other words, I mean that the model below is not the Aligned Xception trained on ImageNet~

> pretrain_dict = model_zoo.load_url('http://data.lip6.fr/cadene/pretrainedmodels/xception-b5690688.pth')

I am wondering whether you have trained the Aligned Xception model on the ImageNet. It would be great if you could share the models~ 

Besides, I am also wondering if the Imagnet pre-trained Aligned Xception model performs better than modified ResNet-101.

**Comment by Gaoyiminggithub** [comment]
Hi,
we have not trained the model on the ImageNet. Our model is trained on the COCO.
Maybe you can get the Xception pretrained on the ImageNet on [here](https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md).

**Comment by PkuRainBow** [comment]
@Gaoyiminggithub It seems that [link ](https://github.com/Cadene/pretrained-models.pytorch) is not trained on COCO. 

 I am wondering whether could you share the checkpoints of the **Aligned** Xception (instead of Xception) trained on COCO.

**Comment by Gaoyiminggithub** [comment]
@PkuRainBow 
Hi, the link 
`https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md`
 I provided above is the checkpoints of the Aligned Xception :) (you can see and find the imagenet and coco pretrained in it)
Maybe you confuse the link.

**Comment by PkuRainBow** [comment]
@Gaoyiminggithub  Thanks for your quick reply, could you provide me the pytorch version or we can load the weights of the tensorflow models (in the Figure as below) directly?

![image](https://user-images.githubusercontent.com/4639578/58335773-e2ed0100-7e74-11e9-835a-85ca8ac24a51.png)

**Comment by Gaoyiminggithub** [comment]
@PkuRainBow 
I could not provide the pytorch version right now, because I delete it after I have trained the model on coco, but maybe I could convert that in next week.
You could convert the tensorflow models to the pytorch version by extracting the weight from the tensorflow models and then fill the pytorch model.weight with it.

**Comment by PkuRainBow** [comment]
@Gaoyiminggithub Thanks for your help. I am wondering if you could share with me the mentioned COCO trained model.

**Comment by Gaoyiminggithub** [comment]
you can download the coco pretrained model in [here](https://drive.google.com/file/d/18WiffKnxaJo50sCC9zroNyHjcnTxGCbk/view?usp=sharing).
The normalization way should follow [here](https://github.com/tensorflow/models/blob/master/research/deeplab/core/feature_extractor.py#L194-L197).

**Comment by PkuRainBow** [comment]
@Gaoyiminggithub Thanks~

**Comment by PkuRainBow** [comment]
@Gaoyiminggithub Why do you comment the BN operation during the "Exit flow" as below?

```
        # Exit flow
        x = self.block20(x)
        x = self.conv3(x)
        # x = self.bn3(x)
        x = self.relu(x)

        x = self.conv4(x)
        # x = self.bn4(x)
        x = self.relu(x)

        x = self.conv5(x)
        # x = self.bn5(x)
        x = self.relu(x)
```

**Comment by Gaoyiminggithub** [comment]
@PkuRainBow because I put the BN operation in self.conv3/4/5.

**Comment by PkuRainBow** [comment]
@Gaoyiminggithub Got it~  Besides, it would be great if you could share with me the converted Pytorch model based on the official ImageNet based tensorflow models.

Besides, I want to check with you that the data normalization is simply to convert all the values to be in the range [-1, 1]?

Thanks a lot!
